File: solutions/0058-length-of-last-word/length-of-last-word.cpp
// Given a string s consists of upper/lower-case alphabets and empty space characters ' ', return the length of last word in the string.
//
// If the last word does not exist, return 0.
//
// Note: A word is defined as a character sequence consists of non-space characters only.
//
// Example:
//
//
// Input: "Hello World"
// Output: 5
//
//
//  
//


class Solution {
public:
    int lengthOfLastWord(string s) {
        if(s.length() == 0) return 0;
        int cnt = 0;
        bool first = 1;
        for(int i=s.length()-1; i>=0; --i) {
            if(first) {
                if(s[i] == ' ') ;
                else {
                    first = 0;
                    ++cnt;
                }
            }
            else if(s[i] != ' ') ++cnt;
            else break;
        }
        return cnt;
    }
};

File: solutions/0058-length-of-last-word/length-of-last-word.java
// Given a string s consists of upper/lower-case alphabets and empty space characters ' ', return the length of last word in the string.
//
// If the last word does not exist, return 0.
//
// Note: A word is defined as a character sequence consists of non-space characters only.
//
// Example:
//
//
// Input: "Hello World"
// Output: 5
//
//
//  
//


class Solution {
    public int lengthOfLastWord(String s) {
        if(s == null || s.length() == 0) return 0;
        int end = s.length() - 1;
        while(end >= 0 && s.charAt(end) == ' ') end--;
        if(end == -1) return 0;
        int start = end;
        while(start >= 0 && s.charAt(start) != ' ') start--;
        return end - start;
    }
}
